i6_models/parts/best_rq/quantizer_with_pca.py

```diff
@@ -0,0 +1,250 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.linalg import vector_norm
+from i6_experiments.users.jxu.experiments.pretrain.pytorch_networks.pca_module import IncrementalPCAonGPU
+
+__all__ = [
+    "PCAProjectionQuantizer",
+]
+
+
+class IncrementalPCAonGPU:
```

Comment:
Why is the GPU aspect relevant? Wouldn't the same code work for any device?

Maybe make `device: torch.device` just an argument to the class?

Also, I see below that you actually also use CPU if GPU is not available...

Also, I wonder, shouldn't this be a Torch module?

```suggestion
class IncrementalPCA(nn.Module):
```